core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java

```diff
@@ -412,6 +418,27 @@ default B putRawQueryParameter(String name, String value) {
 
         ExecutionAttributes executionAttributes();
 
+        /**
+         * Sets the signer to use for signing the request. This signer get priority over the signer set on the client while
+         * signing the requests. If this value is null, then the client level signer is used for signing the request.
+         *
+         * @param signer Signer for signing the request
+         * @return This object for method chaining
+         */
+
+
+        /**
+         * Sets the endpointProvider to use for resolving the endpoint of the request. This endpointProvider gets priority
+         * over the endpointProvider set on the client while resolving the endpoint for  the requests.
+         * If this value is null, then the client level endpointProvider is used for resolving the endpoint.
+         *
+         * @param endpointProvider Endpoint Provider that will override the resolving the endpoint for the request.
+         * @return This object for method chaining
+         */
+        B endpointProvider(EndpointProvider endpointProvider);
```

Comment:
Do we support this? I can't find the code where we use request-level endpoint provider. Did I miss something?